### PR TITLE
Update an example from the System book chapter

### LINKED ISSUE #918: Docs: Example in the book invalid in most cases
The example regarding ActiveCamera is out of date:
https://www.amethyst.rs/book/develop/concepts/system.html#getting-a-component-of-a-specific-entity

It can panic, because no ActiveCamera Resource is found,

### PR DESCRIPTION
**Pull request by Moxinilian**
Fixes #918 and a simple warning in renderer.

### CODE REVIEW

File: book/src/concepts/state.md [hunk 1]
```diff
@@ -183,6 +183,7 @@ If you were to replace `()` by `MyEventType`, then you could react to them from
 # use amethyst::prelude::*;
 # use amethyst::renderer::VirtualKeyCode;
 # use amethyst::input::is_key_down;
+# struct GameplayState;
```
Comment (jojolepro) [inline book/src/concepts/state.md]: https://github.com/amethyst/amethyst/pull/962/files#diff-9c67de6309fc1e25b360604449cc8301R193

what is MyState used for here?

File: book/src/concepts/system.md [hunk 2]
```diff
@@ -66,24 +68,24 @@ Sometimes, it can be useful to get a component in the storage for a specific ent
 
 ```rust,no_run,noplaypen
 # extern crate amethyst;
-# use amethyst::ecs::{System, ReadExpect, WriteStorage};
+# use amethyst::ecs::{Entity, System, WriteStorage};
 # use amethyst::core::Transform;
-# use amethyst::renderer::ActiveCamera;
-struct CameraGoesUp;
+struct WalkPlayerUp {
+    player: Entity,
```
Comment (jojolepro) [inline book/src/concepts/system.md]: This works, but it is a bad practice to store runtime data in systems, as it reduces re-usability.
Would it be possible to have a "MovePlayerUp" component and the system join over (Transform, MoveUp) instead.
MoveUp can have a float inside called move_speed and the system multiplying move_speed by time.delta_seconds() when moving the player.

File: book/src/concepts/state.md (hunk 1 above)
Comment (Moxinilian) [reply book/src/concepts/state.md]: I don't know, git blame you.

File: book/src/concepts/system.md (hunk 2 above)
Comment (Moxinilian) [reply book/src/concepts/system.md]: The point is that the reader does not know join at that point.
And no.

**Pull request merged**